guard/src/commands/reporters/validate/sarif.rs

```diff
@@ -0,0 +1,267 @@
+                    Some(location) => (location.line, location.col),
+                    None => (0, 0),
+                };
+
+                let message = SarifMessage {
+                    text: handle_messages(&messages),
+                };
+
+                let locations = generate_sarif_locations(name, start_line, start_column);
+
+                results.push(SarifResult {
+                    rule_id,
+                    message,
+                    level: String::from("error"),
+                    locations,
+                });
+
+                results
+            })
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone)]
+#[serde(rename_all = "camelCase")]
+struct SarifPhysicalLocation {
+    artifact_location: SarifArtifactLocation,
+    region: SarifRegion,
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone)]
+#[serde(rename_all = "camelCase")]
+struct SarifRegion {
+    start_line: usize,
+    start_column: usize,
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone)]
+#[serde(rename_all = "camelCase")]
+struct SarifLocation {
+    physical_location: SarifPhysicalLocation,
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone)]
+struct SarifRule {
+    id: String,
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct SarifReport {
+    #[serde(rename = "$schema")]
+    schema: String,
+    version: String,
+    runs: Vec<SarifRun>,
+}
+
+impl SarifReport {
+    pub(crate) fn new(reports: &[FileReport<'_>]) -> Self {
+        Self {
+            schema: String::from(SARIF_SCHEMA_URL),
+            version: String::from(SARIF_SCHEMA_VERSION),
+            runs: vec![SarifRun::from(reports)],
+        }
+    }
+}
+
+impl Default for SarifDriver {
+    fn default() -> Self {
+        Self {
+            name: String::from(env!("CARGO_PKG_NAME")),
+            semantic_version: env!("CARGO_PKG_VERSION").to_string(),
+            full_name: format!("{} {}", env!("CARGO_PKG_NAME"), env!("CARGO_PKG_VERSION"),),
+            organization: String::from(ORGANIZATION),
+            download_uri: env!("CARGO_PKG_REPOSITORY").to_string(),
+            information_uri: env!("CARGO_PKG_REPOSITORY").to_string(),
+            short_description: SarifMessage {
+                text: env!("CARGO_PKG_DESCRIPTION").to_string(),
+            },
+        }
+    }
+}
+
+fn handle_messages(messages: &Messages) -> String {
+    format!(
+        "{} {}",
+        messages.error_message.clone().unwrap_or_default(),
+        messages.custom_message.clone().unwrap_or_default()
+    )
+}
+
+fn extract_rule_id(rule_name: &str) -> String {
+    let first_part_of_rule_file_name: Vec<&str> = rule_name.split('.').collect();
+
+    let uppercase_first_part_of_rule_file_name = first_part_of_rule_file_name
+        .first()
+        .map_or(String::default(), |&s| s.to_uppercase());
+
+    uppercase_first_part_of_rule_file_name.to_string()
```

Comment:
Minor: it's unnecessary to call `.to_string()` at the end here. This variable is already a String type. This can be changed to 

```
fn extract_rule_id(rule_name: &str) -> String {
    let first_part_of_rule_file_name: Vec<&str> = rule_name.split('.').collect();

    first_part_of_rule_file_name
        .first()
        .map_or(String::default(), |&s| s.to_uppercase())
}
```